harness/determined/cli/command.py

```diff
@@ -319,9 +359,14 @@ def parse_config_overrides(config: Dict[str, Any], overrides: Iterable[str]) ->
             if key in _CONFIG_PATHS_COERCE_TO_LIST:
                 value = [value]
 
-        # TODO(#2703): Consider using full JSONPath spec instead of dot
-        # notation.
-        config = _set_nested_config(config, key.split("."), value)
+        # Convert config override in dot notation to json path
+        config_arg_in_json = _dot_to_json(key, value)
+        # Convert NoneType value None in config to empty string {}
+        config = _fix_none_value_in_config(config, key.split("."))
+        # Merge two objects in json format
+        config = merge_dicts(config, config_arg_in_json)
+
+    return config
```

Comment:
There was previously no "return" in this function. What is "return config" buying us here?